docs/release-info/release-notes.md

```diff
@@ -57,50 +57,745 @@ For tips about how to write a good release note, see [Release notes](https://git
+- Ensuring that the `arrayIngestMode: array` context parameter is only set when the user opts in to arrays.
+- Setting `arrayIngestMode: array` only if Druid detects that the ingestion spec includes dimensions specs of type `auto` + `castToType: ARRAY<...>`.
+
+![Run panel shows array ingest mode](./assets/30.0.0-run-panel.png)
+
+[#15927](https://github.com/apache/druid/pull/15927)
+
+#### Kafka input format
+
+Improved how the web console determines the input format for a Kafka source.
+Instead of defaulting to the Kafka input format for a Kafka source, the web console now only picks the Kafka input format if it detects any of the following in the Kafka sample: a key, headers, or more than one topic.
+
+[#16180](https://github.com/apache/druid/pull/16180)
+
+#### Improved handling of lookups during sampling
+
+Rather than sending a transform expression containing lookups to the sampler, Druid now substitutes the transform expression with a placeholder.
+This prevents the expression from blocking the flow.
+
+![Change the transform expression to a placeholder](./assets/30.0.0-sampler-lookups.png)
+
+[#16234](https://github.com/apache/druid/pull/16234)
+
 #### Other web console improvements
 
-### Ingestion
+* Added the fields **Avro bytes decoder** and **Proto bytes decoder** for their input formats [#15950](https://github.com/apache/druid/pull/15950)
+* Added support for exporting results for queries that use the MSQ task engine [#15969](https://github.com/apache/druid/pull/15969)
+* Fixed an issue with the [Tasks](https://druid.apache.org/docs/latest/operations/web-console#tasks) view returning incorrect values for **Created time** and **Duration** fields after the Overlord restarts [#16228](https://github.com/apache/druid/pull/16228)
+* Fixed the Azure icon not rendering in the web console [#16173](https://github.com/apache/druid/pull/16173)
+* Fixed the supervisor offset reset dialog in the web console [#16298](https://github.com/apache/druid/pull/16298)
+* Improved the user experience when the web console is operating in manual capabilities mode [#16191](https://github.com/apache/druid/pull/16191)
+* Improved the web console to detect doubles better [#15998](https://github.com/apache/druid/pull/15998)
+* Improved the query timer as follows:
+  * Timer isn't shown if an error happens
+  * Timer resets if changing tabs while query is running
+  * Error state is lost if tab is switched twice
+
+  [#16235](https://github.com/apache/druid/pull/16235)
+* The web console now suggests the `azureStorage` input type instead of the deprecated `azure` storage type [#15820](https://github.com/apache/druid/pull/15820)
+* The download query detail archive option is now more resilient when the detail archive is incomplete [#16071](https://github.com/apache/druid/pull/16071)
+* You can now set `maxCompactionTaskSlots` to zero to top compaction tasks [#15877](https://github.com/apache/druid/pull/15877)
```

Comment:
Typo? Does it mean `to stop` instead of `to top`?

Reply:
Good catch - it should say `to stop`. Updated.